client/src/app/editor/file-outputs/file-outputs.component.ts

```diff
@@ -57,30 +60,64 @@ export class FileOutputsComponent implements OnChanges, OnInit {
   ngOnInit() {
     const outputFileId = this.route.snapshot.queryParamMap.get('preview');
 
-    if (outputFileId) {
-      this.hasOutput
-        .pipe(
-          filter(hasOutput => hasOutput),
-          mergeMap(() => this.dataSource.connect()),
-          mergeMap(outputFiles => outputFiles),
-          filter(outputFile => outputFile.id === outputFileId),
-          filter(outputFile => isImage(outputFile.name)),
-          take(1)
+    const executionSelected$ = this.executionId$.pipe(
+      switchMap(executionId => this.labExecutionService.executionExists(executionId).pipe(catchError(() => of(false)))),
+      filter(Boolean)
+    );
+
+    const executionsAdded$ = this.labExecutionService.observeExecutionsForLab(this.lab).pipe(
+      mergeMap(executions => executions),
+      filter(labExecution => labExecution.id === this.executionId),
+      distinctUntilChanged((a, b) => a.id === b.id),
+      skip(1)
```

Comment:
Why are we skipping the first one here?

Reply:
When you load the page for the first time `observeExecutionsForLab`, which under the hood listens for `child_added`, will emit the entire list of executions. In this case we don't need the value from `observeExecutionsForLab` because `executionSelected$` already emits the execution we want to observe. If we don't skip the first value from `executionsAdded$` we'd end up with the same execution being emitted twice. `executionsAdded$` should only emit if a new execution was started, because `executionSelected$` will not emit in that case.

Does that make sense?

Reply:
Can we add a comment about that then, so that it's clear the next time someone runs over this?